src/middleware/externalRequests.js

```diff
@@ -63,8 +64,213 @@ const prepareRequestConfig = requestDetails => {
+        ctx.status = response.status
+      } else {
+        ctx.secondaryCompleted = true
+
+        setKoaResponseBody(ctx, request, response.data)
+      }
+    }
+  } else {
+    if (request.primary) {
+      setKoaResponseBodyFromPrimary(ctx, request, err.message)
+
+      ctx.primaryReqFailError = true
+      ctx.status = 500
+    } else {
+      ctx.secondaryFailError = true
+
+      setKoaResponseBody(ctx, request, err.message)
+    }
+    error = {message: err.message}
+  }
+
+  return {response, error}
+}
+
+// Sets the koa response body from the primary request's response body
+const setKoaResponseBodyFromPrimary = (ctx, request, body) => {
+  ctx.hasPrimaryRequest = true
+  ctx.body = {}
+  ctx.body[request.id] = body
+}
+
+// Sets the koa response body if there is no primary request
+const setKoaResponseBody = (ctx, request, body) => {
+  if (!ctx.hasPrimaryRequest) {
+    ctx.body[request.id] = body
+  }
+}
+
+const createAxiosConfig = (request, body) => {
+  const axiosConfig = {
+    url: request.url,
+    method: request.method,
+    body: body
+  }
+
+  if (
+    request.credentials &&
+    request.credentials.username &&
+    request.credentials.password
+  ) {
+    axiosConfig.auth = {
+      username: request.credentials.username,
+      password: request.credentials.password
+    }
+  }
+
+  if (request.headers) {
+    axiosConfig.headers = request.headers
+  }
+
+  return axiosConfig
+}
+
+const sendMappedObject = (ctx, axiosConfig, request, body) => {
+  const reqTimestamp = new Date()
+  let response, error, responseTimestamp
+
+  return axios(axiosConfig)
+    .then(resp => {
+      response = resp
+      response.body = resp.data
+      responseTimestamp = new Date()
+
+      if (request.primary) {
+        setKoaResponseBodyFromPrimary(ctx, request, response.body)
+
+        ctx.status = response.status
+      } else {
+        setKoaResponseBodyFromPrimary(ctx, request, response.body)
+      }
+    })
+    .catch(err => {
+      responseTimestamp = new Date()
+
+      const result = handleRequestError(ctx, request, err)
+      response = result.response
+      error = result.error
+    })
+    .finally(() => {
+      if (ctx.request.header && ctx.request.header['X-OpenHIM-TransactionID']) {
+        const orch = createOrchestration(
+          request,
+          body,
+          response,
+          reqTimestamp,
+          responseTimestamp,
+          request.id,
+          error
+        )
+
+        ctx.orchestrations.push(orch)
+      }
+    })
+}
+
+const setStatusText = ctx => {
+  if (ctx.primaryReqFailError) {
+    ctx.statusText = 'Failed'
+  } else if (!ctx.primaryReqFailError && ctx.secondaryFailError) {
+    ctx.statusText = 'Completed with error(s)'
+  } else if (
+    !ctx.primaryReqFailError &&
+    !ctx.secondaryFailError &&
+    (ctx.primaryCompleted || ctx.secondaryCompleted)
+  ) {
+    ctx.statusText = 'Completed'
+  } else {
+    ctx.statusText = 'Successful'
+  }
```

Comment:
@bradsawadye This logic can also be moved to the orchestrator file

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/36/commits/cb529d5a7dd6a90f4a87e0ca2f75d414071b77df